feat!: execution payload refactor

Thunkar opened the pull request:
Part of: https://linear.app/aztec-labs/issue/F-144/improve-interaction-composability

This PR moves `ExecutionPayload` to `stdlib`, cementing its role as the minimal unit of execution on the network. 

As a part of this, the `feePayer` for an execution payload (IF it includes a payment method, that is) is now embedded in the execution payload, since previously it was specified after the fact which was confusing and prone to errors. Finally, renamed a few fee-handling functions in the wallet as discussed with @benesjan in the past.

benesjan commented on boxes/boxes/vanilla/app/embedded-wallet.ts:
```diff
@@ -59,25 +63,25 @@ export class EmbeddedWallet extends BaseWallet {
+   * @returns - Complete fee options that can be used to create a transaction execution request
    */
-  override async getDefaultFeeOptions(
+  override async completeFeeOptions(
```
Very nice name! Gracias 😆 😍

benesjan commented on yarn-project/stdlib/src/tx/execution_payload.ts:
```diff
@@ -0,0 +1,57 @@
+    public capsules: Capsule[],
+    /** Extra hashed values to be injected in the execution cache */
+    public extraHashedArgs: HashedValues[] = [],
+    /** The address that is paying for the fee in this execution payload (if any) */
```
Could you add here a note what happens if fee payer is undefined?

Thunkar replied:
Done!

benesjan commented on yarn-project/stdlib/src/tx/execution_payload.ts:
```diff
@@ -0,0 +1,57 @@
+  }
+
+  // Use the first non-null fee payer
+  const feePayer = feePayers.length > 0 ? feePayers[0] : undefined;
```
```suggestion
  // Collect unique fee payers
  const uniqueFeePayers = new Set(
    requests
      .map(r => r.feePayer)
      .filter((fp): fp is AztecAddress => fp !== undefined)
      .map(fp => fp.toString()),
  );

  if (uniqueFeePayers.size > 1) {
    throw new Error(
      `Cannot merge execution payloads with different fee payers. Found: ${Array.from(uniqueFeePayers).join(', ')}`,
    );
  }

  const feePayer = uniqueFeePayers.size == 1 ? AztecAddress.fromString(Array.from(uniqueFeePayers)[0]) : undefined;
```

Think it's cleaner here to just get the unique non-null fee payers and then operate on that.

benesjan reviewed (approved):
Very nice! Thanks

Thunkar committed:
feat!: execution payload refactor

Part of: https://linear.app/aztec-labs/issue/F-144/improve-interaction-composability

This PR moves `ExecutionPayload` to `stdlib`, cementing its role as the minimal unit of execution on the network.

As a part of this, the `feePayer` for an execution payload (IF it includes a payment method, that is) is now embedded in the execution payload, since previously it was specified after the fact which was confusing and prone to errors. Finally, renamed a few fee-handling functions in the wallet as discussed with @benesjan in the past.

Co-authored-by: Gregorio Juliana <gregojquiros@gmail.com>
Co-authored-by: thunkar <gregojquiros@gmail.com>
